app/org/sagebionetworks/bridge/models/accounts/ConsentStatus.java

```diff
@@ -0,0 +1,112 @@
+package org.sagebionetworks.bridge.models.accounts;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.sagebionetworks.bridge.models.subpopulations.SubpopulationGuid;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class ConsentStatus {
+
+    public static ConsentStatus forSubpopulation(List<ConsentStatus> statuses, SubpopulationGuid subpopGuid) {
+        for (int i=0; i < statuses.size(); i++) {
+            if (statuses.get(i).getSubpopulationGuid().equals(subpopGuid.getGuid())) {
+                return statuses.get(i);
+            }
+        }
+        return null;
+    }
+
+    public static boolean isUserConsented(List<ConsentStatus> statuses) {
+        checkNotNull(statuses);
+        return !statuses.isEmpty() && statuses.stream().allMatch(status -> {
+            return !status.isRequired() || status.isConsented();
+        });
+    }
+
+    /**
+     * Are all the required consents up-to-date?
+     * @return
+     */
+    public static boolean isConsentCurrent(List<ConsentStatus> statuses) {
+        checkNotNull(statuses);
+        return !statuses.isEmpty() && statuses.stream().allMatch(status -> {
+            return !status.isRequired() || status.isMostRecentConsent();   
+        });
+    }
+    
+    public static boolean hasOnlyOneSignedConsent(List<ConsentStatus> statuses) {
+        checkNotNull(statuses);
+        int count = 0;
+        for (ConsentStatus status : statuses) {
+            if (status.isConsented()) {
+                count ++;
+            }
+        }
+        return !statuses.isEmpty() && count == 1;
+    }
+    
+    private final String name;
+    private final String subpopulationGuid;
+    private final boolean required;
+    private final boolean consented;
+    private final boolean mostRecentConsent;
+    
+    @JsonCreator
+    public ConsentStatus(@JsonProperty("name") String name, @JsonProperty("subpopulationGuid") String subpopulationGuid,
+            @JsonProperty("required") boolean isRequired, @JsonProperty("consented") boolean isConsented, 
+            @JsonProperty("mostRecentConsent") boolean isMostRecentConsent) {
+        this.name = checkNotNull(name);
+        this.subpopulationGuid = checkNotNull(subpopulationGuid);
+        this.required = isRequired;
+        this.consented = isConsented;
+        this.mostRecentConsent = isMostRecentConsent;
+    }
```

Comment:
Minor: We have two strings and three booleans. This is almost certainly more readable using the Builder pattern.